config/routes.rb

```diff
@@ -92,9 +92,11 @@
   namespace :admin do
     resources :memberships, only: %i[create]
     resources :organizations, only: %i[update]
-    resources :invoices do
+    resources :invoices, except: :show do
       post :regenerate, on: :member
     end
+
+    get '/invoices/:id', to: 'invoices#show' if Rails.env.development?
```

Comment:
What about having a `dev_tools` namespace in the routes?
I think it's better to not mix the concerns. It could be used in the future to add any other rendering logic, like credit notes, or email templates.
What do you think?

Reply:
That's a good point, I'll change the route.